### PR TITLE
feat: add d08 validation effective date must match start date

### PR DESCRIPTION
**Pull request by Mech0z**
## References

* #1765

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs [hunk 1]
```diff
@@ -209,6 +209,10 @@ public static class CimValidationErrorTextTemplateMessages
         public const string MonthlyPriceSeriesEndDateMustBeFirstOfMonthErrorText =
             "The monthly price series with transaction ID {{ChargeOperationId}} for charge ID {{DocumentSenderProvidedChargeId}} of type {{ChargeType}} owned by {{ChargeOwner}} has an end time interval that does not match the first of a month or the charge's stop date";
 
+        [ErrorMessageFor(ValidationRuleIdentifier.EffectiveDateMustMatchStartDate)]
+        public const string EffectiveDateMustMatchStartDateErrorText =
+            "Start date and effective date must have the same value for charge with ID {{DocumentSenderProvidedChargeId}} of type {{ChargeType}} owned by {{ChargeOwner}}";
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs:214]:
```suggestion
            "Price series start interval and effective date must have the same value for charge with ID {{DocumentSenderProvidedChargeId}} of type {{ChargeType}} owned by {{ChargeOwner}}";
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/Validation/ValidationRuleIdentifier.cs [hunk 2]
```diff
@@ -62,5 +62,6 @@ public enum ValidationRuleIdentifier
         ChargeTypeTariffTaxIndicatorOnlyAllowedBySystemOperator = 51, // VR508 / E0I
         UpdateTaxTariffOnlyAllowedBySystemOperator = 52, // VR508 / E0I
         MonthlyPriceSeriesEndDateMustBeFirstOfMonthOrEqualChargeStopDate = 53, // VR923 / D14
+        EffectiveDateMustMatchStartDate = 54, // VR924 / E0H
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/Validation/ValidationRuleIdentifier.cs:65]:
```suggestion
        EffectiveDateMustMatchPriceSeriesStartInterval = 54, // VR924 / E0H
```

File: docs/business-processes/asynchronous-validations.md [hunk 3]
```diff
@@ -48,6 +48,7 @@ The following asynchronous validation rules are currently implemented in the cha
 |VR.921|Tax indicator must be set to false for a fee|D14|✓|||
 |VR.922|The identification of a charge operation consists of maximal 36 characters|E86|✓|✓||
 |VR.923|The monthly price series end date must be 1st of month, unless it matches the charge stop date.|D14||✓||
+|VR.924|Start date and effective date must have the same value.|E0H||✓||
```
Comment (prtandrup) [inline docs/business-processes/asynchronous-validations.md:51]:
```suggestion
|VR.924|Price series start interval and effective date must have the same value|E0H||✓||
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/Factories/ChargePriceOperationInputValidationRulesFactory.cs [hunk 4]
```diff
@@ -64,6 +64,7 @@ private IEnumerable<IValidationRuleContainer> GetRulesForOperation(ChargePriceOp
                 CreateRuleContainer(new ResolutionTariffValidationRule(operation), operation.OperationId),
                 CreateRuleContainer(new ResolutionFeeValidationRule(operation), operation.OperationId),
                 CreateRuleContainer(new ResolutionIsRequiredRule(operation), operation.OperationId),
+                CreateRuleContainer(new EffectiveDateMustMatchStartDateRule(operation), operation.OperationId),
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/Factories/ChargePriceOperationInputValidationRulesFactory.cs:67]:
```suggestion
                CreateRuleContainer(new EffectiveDateMustMatchPriceSeriesStartInterval(operation), operation.OperationId),
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs [hunk 5]
```diff
@@ -209,6 +209,10 @@ public static class CimValidationErrorTextTemplateMessages
         public const string MonthlyPriceSeriesEndDateMustBeFirstOfMonthErrorText =
             "The monthly price series with transaction ID {{ChargeOperationId}} for charge ID {{DocumentSenderProvidedChargeId}} of type {{ChargeType}} owned by {{ChargeOwner}} has an end time interval that does not match the first of a month or the charge's stop date";
 
+        [ErrorMessageFor(ValidationRuleIdentifier.EffectiveDateMustMatchPriceSeriesStartInterval)]
+        public const string EffectiveDateMustMatchStartDateErrorText =
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs:213]:
```suggestion
        public const string EffectiveDateMustMatchPriceSeriesStartIntervalErrorText =
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/EffectiveDateMustMatchStartDateRuleTests.cs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+using GreenEnergyHub.TestHelpers;
+using NodaTime;
+using NodaTime.Extensions;
+using Xunit;
+using Xunit.Categories;
+
+namespace GreenEnergyHub.Charges.Tests.Domain.Dtos.ChargePriceCommands.Validation.InputValidation.ValidationRules
+{
+    [UnitTest]
+    public class EffectiveDateMustMatchStartDateRuleTests
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/EffectiveDateMustMatchStartDateRuleTests.cs:30]:
```suggestion
    public class EffectiveDateMustMatchPriceSeriesStartIntervalRuleTests
```

**Review by prtandrup — APPROVED** [review]
Approved with a few renaming comments

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/Factories/ChargePriceOperationInputValidationRulesFactory.cs (hunk 4 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargePriceCommands/Validation/InputValidation/Factories/ChargePriceOperationInputValidationRulesFactory.cs]: renamed to EffectiveDateMustMatchPriceSeriesStartIntervalRule

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs (hunk 5 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure.Core/Cim/ValidationErrors/CimValidationErrorTextTemplateMessages.cs]: renamed

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/EffectiveDateMustMatchStartDateRuleTests.cs (hunk 6 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargePriceCommands/Validation/InputValidation/ValidationRules/EffectiveDateMustMatchStartDateRuleTests.cs]: done

**Pull request merged**